tutorials/hyperspectral/Advanced_Spectral_Signatures.Rmd

```diff
@@ -0,0 +1,291 @@
+names(b)[2]=paste0("Point_",i)
+
+# add reflectance values for this pixel to our combined data.frame called Pixel_df
+Pixel_df=cbind(Pixel_df,b[2])
+}
+
+```
+
+## Plot spectral signatures using ggplot2
+Finally, we have everything that we need to plot the spectral signatures for 
+each of the pixels that we clicked. In order to color our lines by the different
+land cover types, we will first reshape our data using the `melt` function, then
+plot the spectral signatures.
+
+```{r plot-spectral-signatures}
+# Use the melt() funciton to reshape the dataframe into a format that ggplot prefers
+Pixel.melt=melt(Pixel_df, id.vars = "wavelengths", value.name = "Reflectance")
+
+ggplot()+
+  geom_line(data = Pixel.melt, mapping = aes(x=wavelengths, y=Reflectance, color=variable), lwd=1.5)+
+  scale_colour_manual(values = c("green2", "green4", "grey50","tan4","blue3"),
+                      labels = c("Field", "Tree", "Roof","Soil","Water"))+
+  labs(color = "Cover Type")+
+  ggtitle("Land cover spectral signatures")+
+  theme(plot.title = element_text(hjust = 0.5, size=20))+
+  xlab("Wavelength")
+```
+
+Nice! However, there seems to be something weird going on in the wavelengths near 1400nm and 1850 nm...
+
+## Atmospheric Absorbtion Bands 
+Those irregularities around 1400nm and 1850 nm are two major atmospheric absorbtion bands - regions
+where gasses in the atmosphere (primarily carbon dioxide and water vapor) absorb radiation, and 
+therefore, obscure the reflected radiation that the imaging spectrometer measures. Fortunately, the 
+lower and upper bound of each of those atmopheric absopbtion bands is specified in the HDF5 file. 
+Let's read those bands and plot rectangles where the reflectance measurements are obscured by 
+atmospheric absorbtion. 
+
+```{r mask-atmospheric-absorbtion-bands}
+# grab Reflectance metadata (which contains absorption band limits)
+reflMetadata<- h5readAttributes(f,"/SJER/Reflectance" )
+
+ab1 = reflMetadata$Band_Window_1_Nanometers
+ab2 = reflMetadata$Band_Window_2_Nanometers
+
+## Plot spectral signatures again with rectangles showing the absorption bands
+ggplot()+
+  geom_line(data = Pixel.melt, mapping = aes(x=wavelengths, y=Reflectance, color=variable), lwd=1.5)+
+  geom_rect(mapping=aes(ymin=min(Pixel.melt$Reflectance),ymax=max(Pixel.melt$Reflectance), xmin=ab1[1], xmax=ab1[2]), color="black", fill="grey40", alpha=0.8)+
+  geom_rect(mapping=aes(ymin=min(Pixel.melt$Reflectance),ymax=max(Pixel.melt$Reflectance), xmin=ab2[1], xmax=ab2[2]), color="black", fill="grey40", alpha=0.8)+
+  scale_colour_manual(values = c("green2", "green4", "grey50","tan4","blue3"),
+                      labels = c("Field", "Tree", "Roof","Soil","Water"))+
+  labs(color = "Cover Type")+
+  ggtitle("Land cover spectral signatures")+
+  theme(plot.title = element_text(hjust = 0.5, size=20))+
+  xlab("Wavelength")
+```
+
+Now we can clearly see that the noisy sections of each spectral signature is within the atmospheric
```

Comment:
Grammer: "sections" "are" or "section" "is"  - not sure which you want - I think "sections".

Reply:
Yes, sections are. Corrected here. Good eye!